File: leet/blind75/invertBinaryTree.js
const { treeGen, TreeNode } = require('../viewLib/forest.js')

function invertTree(root) {
  if(root == null) {
    return root
  }
  let temp = root.left
  root.left = root.right
  root.right = temp
  invertTree(root.left)
  invertTree(root.right)
  return root
}

let vals = [4,2,7,1,3,6,9]
let nodes = []
for(let val of vals) {
  nodes.push(val == null ? null : new TreeNode(val))
}

let tree = treeGen(nodes)
console.log(tree)
invertTree(tree)
console.log(tree)

File: leet/invertBinaryTree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def invertTree(root: TreeNode) -> TreeNode:
    if root is None:
        return None
    temp = root.right
    root.right = root.left
    root.left = temp
    invertTree(root.left)
    invertTree(root.right)
    return root
